cordyceps/src/stack.rs

```diff
@@ -400,6 +416,245 @@ impl<T> Default for Links<T> {
+                            // As above, if cursor.next > element, then we
+                            // need to insert between cursor and next.
+                            //
+                            // First, get the current element's links...
+                            let elinks = T::links(ptr).as_mut();
+                            // ...then store cursor.next.next in element.next,
+                            // and store element in cursor.next.
+                            elinks.next.with_mut(|enext| {
+                                *enext = mutref.replace(ptr);
+                            });
+                            // If we have placed element, there is no next value
+                            // for cursor.
+                            None
+                        } else {
+                            // If cursor.next <= element, then we just need to
+                            // iterate, so return the NonNull that represents
+                            // cursor.next, so we can move cursor there.
+                            Some(nptr)
+                        }
+                    } else {
+                        // "just append" case - assign element to cursor.next
+                        *mutref = Some(ptr);
+                        // If we have placed element, there is no next value
+                        // for cursor
+                        None
+                    }
+                });
+
+                // We do the assignment through this tricky return to ensure that the
+                // mutable reference to cursor.next we held as "mutref" above has been
+                // dropped, so we are not mutating `cursor` while a reference derived
+                // from it's provenance is live.
+                //
+                // We also can't early return inside the loop because all of the body
+                // is inside a closure.
+                //
+                // This might be overly cautious, refactor carefully (with miri).
+                let Some(n) = next else {
+                    // We're done, return
+                    return;
+                };
+                cursor = n;
+            }
+        }
+    }
+
+    /// Iterate through the items of the list, in sorted order
+    pub fn iter(&self) -> SortedListIter<'_, T> {
+        SortedListIter {
+            _plt: PhantomData,
+            node: self.head,
+        }
+    }
+}
+
+impl<T: Linked<Links<T>>> Extend<T::Handle> for SortedList<T> {
+    fn extend<I: IntoIterator<Item = T::Handle>>(&mut self, iter: I) {
+        for elem in iter {
+            self.insert(elem);
+        }
+    }
+}
+
+impl<T: Linked<Links<T>>> Drop for SortedList<T> {
+    fn drop(&mut self) {
+        // TODO: This is a hack. We just turn the list into a stack then run the
+        // stack drop code. It already has correct + tested logic for dropping
+        // a singly linked list of items one at a time.
```

Comment:
I actually think this is the Correct Thing To Do Here, so I'm unsure if it really needs a TODO comment.

Reply:
Yep, will weaken. Didn't know if we wanted to have some common drop fn instead to make the intent more clear, but I agree it doesn't make sense to impl the functionality twice.